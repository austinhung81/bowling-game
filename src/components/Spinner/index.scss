.spinner {
  display: flex;
  align-items: center;
  justify-content: center;
}

.spinner::before {
  content: "";
  display: inline-block;
  width: 50px;
  height: 50px;
  border: 3px solid rgba(0, 0, 0, 0.12);
  border-radius: 50%;
  border-top-color: #0547cd;
  animation: spin 1s ease-in-out infinite;
}

@keyframes spin {
  to {
    transform: rotate(360deg);
  }
}