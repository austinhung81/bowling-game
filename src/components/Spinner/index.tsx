import React from "react";
import "./index.scss";

const Spinner = (): JSX.Element => {
  return (
    <div className="spinner" />
  );
};

export default Spinner;