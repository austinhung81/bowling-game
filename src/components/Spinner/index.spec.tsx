
import React from 'react';
import { create } from "react-test-renderer";
import Spinner from './index';

describe('test Spinner component', () => {
 it('should be defined', () => {
   expect(Spinner).toBeDefined();
 });
 it('should match snapshot', () => {
  expect.assertions(1);
  const spinner = create(<Spinner />).toJSON();
  expect(spinner).toMatchSnapshot();
 });
});