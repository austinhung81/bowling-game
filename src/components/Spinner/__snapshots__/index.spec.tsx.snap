// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Spinner component should match snapshot 1`] = `
<div
  className="spinner"
/>
`;
